addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1861,7 +1861,11 @@ const ListUserValueWidget = UserValueWidget.extend({
         'click we-button.o_we_list_add_existing': '_onAddExistingItemClick',
         'click we-select.o_we_user_value_widget.o_we_add_list_item': '_onAddItemSelectClick',
         'click we-button.o_we_checkbox_wrapper': '_onAddItemCheckboxClick',
-        'change table input': '_onListItemChange',
+        'change table input:not(input[type="number"])': '_onListItemChange',
+        // Numerical inputs trigger the change event when the user presses keyup
+        // or keydown, so the change of input's value should only be taken into
+        // consideration when blurring.
+        'blur table input[type="number"]': '_onListItemChange',
```

Comment:
The lists can now be used with tab key. we-list update the DOM at each input and update the DOM + the editor when the input is blurred and the element that has the focus is not an input in the list. However, I've been looking at how I could make we-lists use we-inputs and it doesn't seem easy. I think the time to do this will be too long.